File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans=0
        left=0
        stack=set()

        for i in range(len(s)):
            while s[i] in stack:
                stack.remove(s[left])
                left+=1
            stack.add(s[i])
            ans=max(ans,i-left+1)
        return ans

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let left=0
    let ans=0
    let set = new Set()
    for(let i=0;i<s.length;i++){
        while(set.has(s[i])){
            set.delete(s[left])
            left++
        }
        set.add(s[i])
        if (ans<(i-left+1)){
            ans=i-left+1
        }
    }     
    return ans
};